src/wavm/mpi.cpp

```diff
@@ -399,6 +585,47 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
     return MPI_SUCCESS;
 }
 
+/**
+ * Waits for all given communications to complete
+ * TODO not implemented
+ */
+WAVM_DEFINE_INTRINSIC_FUNCTION(env,
+                               "MPI_Waitall",
+                               I32,
+                               MPI_Waitall,
+                               I32 count,
+                               I32 requestArray,
+                               I32 statusArray)
+{
+    faabric::util::getLogger()->debug(
+      "S - MPI_Waitany {} {} {}", count, requestArray, statusArray);
+
+    throw std::runtime_error("MPI_Waitall is not implemented!");
+
+    return MPI_SUCCESS;
+}
+
+/**
+ * Waits for any specified send or receive to complete
+ * TODO not implemented
+ */
+WAVM_DEFINE_INTRINSIC_FUNCTION(env,
+                               "MPI_Waitany",
+                               I32,
+                               MPI_Waitany,
+                               I32 count,
+                               I32 requestArray,
+                               I32 index,
+                               I32 status)
+{
+    faabric::util::getLogger()->debug(
+      "S - MPI_Waitany {} {} {} {}", count, requestArray, index, status);
```

Comment:
Is `index` a keyword for something? It's highlighted in this diff but I can't tell. Maybe use `idx` to be safe